src/ws/cockpitauth.c

```diff
@@ -642,22 +550,25 @@ create_creds_for_spawn_authenticated (CockpitAuth *self,
    * passing it back and forth possibly leaking it.
    */
 
-  if (g_str_equal (sl->auth_type, "basic"))
-    password = parse_basic_auth_password (sl->authorization, NULL);
+  if (g_str_equal (ad->auth_type, "basic") ||
+      g_str_equal (ad->auth_type, ACTION_SSH))
+    {
+      password = parse_basic_auth_password (ad->authorization, NULL);
```

Comment:
This path is not taken during testing. Is there a way to have it be taken, or is that hopeless, due to the fact that we mock out the auth?